### ISSUE TITLE
Keys with a colon leads to bad signatures

### ISSUE THREAD

**Issue by carlosmn** [rlmcpherson/s3gof3r#110]
I wanted to have a colon in one of my S3 keys, but it looks like s3gof3r doesn't like them in the keys. Once I removed it from the key it worked fine, but including something like `v0/2:abcde` in the key returns a signature error from S3.

```
gof3r error: 403: "The request signature we calculated does not match the signature you provided. Check your key and signing method."
```

This hapens both from the gof3r tool itself as well as trying to use this as a library in my own tool.

The colon is mentioned in the [naming guidelines](http://docs.aws.amazon.com/AmazonS3/latest/dev/UsingMetadata.html) as "Characters That Might Require Special Handling". Unfortunately the wording there makes it sound like they won't quite tell you how they expect to see them.

**Comment by BogdanSorlea** [comment]
yep.
https://github.com/rlmcpherson/s3gof3r/issues/112

**Comment by rlmcpherson** [comment]
Fixed by #113

**Comment by ewold** [comment]
Also seeing this problem with the plus character Plus ("+")
For example a file with key: 5-ID+Rub+IgG+II+30221_01_2002_FR.docx

```
gof3r error: 403: "The request signature we calculated does not match the signature you provided. Check your key and signing method."
````
